gio-subclass/src/application.rs

```diff
@@ -29,7 +29,20 @@ impl ArgumentList{
         }
     }
 
-    fn remove(){
+    // remove the item at index `idx` and shift the raw array
+    pub fn remove(&mut self, idx: usize){
+        self.items.remove(idx);
+
+        unsafe{
+            let n_args = glib_ffi::g_strv_length(*self.ptr);
+            assert!((idx as u32) < n_args);
```

Comment:
You probably want to put an assertion to this effect at the top of this function before doing anything.

Also the assumption here is in addition that both arrays are matchnig all the time. So an assertion that checks that both lengths are the same could make sense in addition